lecture/views.py

```
@@ -86,20 +86,68 @@ def lecture_pdf(request: HttpRequest,slug:str,pdf_id: int) -> HttpResponse:
         form = LectureForm(request.POST)  # Pass the user if needed
-
+        course = get_object_or_404(Course, slug=slug)
         if form.is_valid():
-            saved_lecture = form.save()
+            saved_lecture = form.save(commit=False)
+            saved_lecture.course=course
+            saved_lecture.save()
             messages.success(request, "Lecture created successfully!")
-            return redirect("lecture_home", slug=saved_lecture.slug)  # Redirect to a relevant page
+            return redirect("lecture_home", slug=slug)  # Redirect to a relevant page
         else:
```

Ruff (0.8.2)

Unnecessary `else` after `return` statement

Remove unnecessary `else`

(RET505)